### DISCUSSION TITLE
2D lighting

### DISCUSSION THREAD

**Discussion by leo6dev** [bevyengine/bevy#4756] [Q&A] [answered]
I hope this is the right spot dose bevy support 2D lighting ? There appears to be 3D support in bevy_pbr and [unity supports 2D lighting](https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@7.1/manual/LightTypes.html#freeform). I am trying change the global lighting in a 2D game as well as adding some spot lights

**Comment by NHodgesVFX** [comment]
I don't think it does

**Comment by cart** [comment, marked as answer]
We don't yet support 2d lighting, but I'd definitely like to at some point. While you wait for a first party solution, this could be accomplished as a 3rd party plugin by writing custom shaders and adding RenderApp systems to extract, process, and queue the relevant 2d data. Bevy's rendering is very modular + configurable.

**Reply by leo6dev** [reply]
Thank you

**Comment by zaycev** [comment]
I've built a prototype for top-down 2D games which works exactly as @cart explained: https://github.com/zaycev/bevy-magic-light-2d, it's not production ready, but I'm open for PRs :)

**Reply by ryanpeach** [reply]
Just so readers are aware, this seems to be very specialized for @zaycev's game. However, it looks like a lot of good work has gone into it too. It could be a good jumping off point.

**Comment by jgayfer** [comment]
I recently published [`bevy_light_2d`](https://github.com/jgayfer/bevy_light_2d) in an attempt to fill this void.
